src/main/kotlin/org/rust/lang/core/types/infer/TypeInference.kt

```diff
@@ -971,9 +971,22 @@ class RsFnInferenceContext(
 
     private fun inferStructTypeArguments(literal: RsStructLiteral, typeParameters: Substitution) {
         literal.structLiteralBody.structLiteralFieldList.mapNotNull { field ->
-            field.expr?.let { expr ->
-                val fieldType = field.type
-                expr.inferTypeCoercableTo(fieldType.substitute(typeParameters))
+            val fieldType = field.type.substitute(typeParameters)
+
+            if (field.expr != null) {
+                field.expr!!.inferTypeCoercableTo(fieldType)
+            } else {
+
+                // TODO: Unify the types of the struct field and the matching declaration
+                //   At this point, the type inference system is looking at a struct
+                //   field that does not have a colon and expr, meaning the shorthand
+                //   form was used. This means we must attempt to find an in-scope match
+                //   for the struct field's name, then infer that its declaration
+                //   is coercible to the field's type.
+
+                // How can I look up a variable that's in scope and shares a name with
+                // the struct field (the identifier below)?
+                val ident = field.identifier
```

Comment:
I can imagine this solution. Note that I didn't tried to even compile it =)
```kotlin
val path = RsCodeFragmentFactory(project).createPath(field.referenceName, literal)
val ty = (resolvePath(path, lookup) as? RsPatBinding)?.let { bindings[it] } ?: TyUnknown
tryCoerce(ty, fieldType)
```

Reply:
Thanks for the input! That general approach worked great.